File: solutions/cpp/collatz-conjecture/1/collatz_conjecture.cpp
#include "collatz_conjecture.h"

namespace collatz_conjecture {
    int steps(int number) {
        if (number <= 0) throw std::domain_error("error");
        int s = 0;
        while (number > 1) {
            if (number % 2 ==0) number /= 2;
            else number = 3 * number + 1;
            s++;
        }
        return s;
    }
}  // namespace collatz_conjecture

File: solutions/python/collatz-conjecture/1/collatz_conjecture.py
def steps(number):
    if number <= 0:
        raise ValueError("Value Error")
    s=0
    while(number!=1):
        s+=1
        if number % 2 == 0:
            number /= 2
        else:
            number = 3 * number + 1
        #print("number=",number)
    return(s)
print(steps(1000000))